### PR TITLE
Created the Game logic

### PR DESCRIPTION
**Pull request by gitnyasha**
Added the code of the logic of the game tic tac toe

### CODE REVIEW

**Review by Forison — CHANGES_REQUESTED** [review]
## status Require:changes🛑
# 🙋‍♂️Team,
you have done a good job on this one so far.
The behavior of the program is perfect but there is one technical issue we have  to address to get this project approved
* [ ] No puts/print and get.chomp outside main.rb file.([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
## [ Optional ]
the classes in your logic file could be placed in two separate files and place in```/lib```. 
## That is it, kindly make this change and submit the project for review.🙂
#### Happy coding💻
#### Addo Forison

**Comment by gitnyasha** [comment]
where is the puts/prints in the logic file?

**Comment by gitnyasha** [comment]
> ## status Require:changes🛑
> # 🙋‍♂️Team,
> you have done a good job on this one so far.
> The behavior of the program is perfect but there is one technical issue we have to address to get this project approved
> 
> * [x]  No puts/print and get.chomp outside main.rb file.([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
> 
> ## [ Optional ]
> the classes in your logic file could be placed in two separate files and place in`/lib`.
> 
> ## That is it, kindly make this change and submit the project for review.🙂
> #### Happy coding💻
> #### Addo Forison

where did I use puts/print in the logic file?

File: bin/main [hunk 1]
```diff
@@ -1,51 +1,77 @@
...
-    break
+
+def game_board(board)
+  puts " #{board.board[0]} | #{board.board[1]} | #{board.board[2]}"
+  puts "-------------"
+  puts " #{board.board[3]} | #{board.board[4]} | #{board.board[5]}"
+  puts "-------------"
+  puts " #{board.board[6]} | #{board.board[7]} | #{board.board[8]}"
+  puts "-------------"
+end
```
Comment (euqueme) [inline bin/main:14]: - [ ] keep your code DRY (don't repeat yourself) this method is repetitive  in several parts so you could use a loop to accomplish the same with fewer lines

File: bin/main [hunk 2]
```diff
@@ -1,51 +1,77 @@
...
+        puts "#{two.name} has won!"
+      end
+      break
+    end
+    unless board.draw?
+      puts 'Its a Draw'
+      break
+    end
+  end
+end
```
Comment (euqueme) [inline bin/main]: - [ ] The code should respect the single responsibility rule for methods (i.e. each method does exactly one thing).

the **game_is_running** method doesn't respect the single responsibility rule so break it down in smaller methods or in fact, I think you can call all of its content directly in the loop you have at line #72

**Review by euqueme — CHANGES_REQUESTED** [review]
> ### Pull Request under the review by Maria Eugenia Queme P. (TSE)

### Hello @gitnyasha :handshake: 

### Thank you for doing the changes requested by the previous reviewer, and thank you for having such a descriptive readme file at this point :medal_sports: 

## Follow up

#### Behaviour

- [x] it lets you leave the players names blank, so let's create a validation to assign a default name if it's left blank or force the player to write a name.

- [x] when the game finds an invalid input it should inform the player about it and asks for their move again, instead it displays 'Already taken' and asks for the move again, so let's create the right validation with a descriptive message for this scenario:

image

- [x] when the first player wins it shows the second player as the winner, actually is displaying the name of player 2 as a winner all the time even if the game is a tie

image
image

#### Implementation details

- [x] I have commented your code bellow with my requirements as a follow up of the changes you made in your main file :point_down: 

## [OPTIONAL]

- [x] [OPTIONAL] Also, I am very sorry this was pointed out as optional by the previous reviewer but having 2 or more classes in your **/lib** folder is actually a hard requirement. perhaps you could create a class for your players?,  I strongly recommend to do this change because it will make your project look more professional and object-oriented.

### That's it! :raised_hands: 

### Please :pray:  do the changes :memo: and submit back for a code review when you are done :rocket: 

Happy Coding! :woman_technologist: 

PROJECT STATUS: REQUESTED CHANGES :vertical_traffic_light:

**Comment by Anna-Myzukina** [comment]
## Hi @gitnyasha  ! Your project looks good :+1:
## Status [Approved]
#### Go ahead and merge to master.
* Good job on rectifying the issues raised in the previous reviews  :100: 

_I hope that in the next projects you can make your PR more descriptive. Tell us what you have done in this PR. What features does your project cover._




#### Goodluck in your next project Happy coding <g-emoji class="g-emoji" alias="computer" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/1f4bb.png">computer</g-emoji>

**Review by Anna-Myzukina — APPROVED** [review]

**Pull request merged**